File: Part 2 - Regression/Simple Linear Regression/Simple_reg.R
#importing data
data=read.csv("Salary_Data.csv")

#splitting train and test set
library(caTools)
Split=sample.split(data$Salary,SplitRatio = 0.75)
train_set=subset(data,Split==TRUE)
test_set=subset(data,Split==FALSE)

#model building
reg=lm(formula =Salary ~ YearsExperience,data = train_set)
summary(reg)

#prediction
y_pred=predict(reg,newdata = test_set)

#visualising data
install.packages("ggplot2")
library(ggplot2)

#on traing data
ggplot()+
  geom_point(aes(train_set$YearsExperience,train_set$Salary),colour="red")+
  geom_line(aes(train_set$YearsExperience,predict(reg,newdata=train_set)),
            colour="blue")+
  ggtitle("Sal vs Exp(traingSet")+
  xlab("Years of Expr")+
  ylab("Salary")

#on test set
ggplot()+
  geom_point(aes(test_set$YearsExperience,test_set$Salary),colour="blue")+
  geom_line(aes(train_set$YearsExperience,predict(reg,newdata=train_set)),
             colour="red")+
  ggtitle("Sal vs Exp(testSet")+
  xlab("Years of Expr")+
  ylab("Salary")
              

File: Part 2 - Regression/Simple Linear Regression/Simple_Reg.py
#importing Lib
import pandas as pd
import numpy as ny
import matplotlib.pyplot as plt

#importing data 
data=pd.read_csv("Salary_Data.csv")
X=data.iloc[:,:-1]
y=data.iloc[:,1:]

#splitting train and testset
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.25,random_state=0)

#model building
from sklearn.linear_model import LinearRegression
reg=LinearRegression()
reg.fit(X_train,y_train)

#prediction on test set
y_pred=reg.predict(X_test)

#visualising the training data
plt.scatter(X_train,y_train,color='red')
plt.plot(X_train,reg.predict(X_train),color="blue")
plt.title("Sal vs Yrs Exp(trainset)")
plt.ylabel("Salary")
plt.xlabel("Years Of Exp")
plt.show()

#visualising the test data
plt.scatter(X_test,y_test,color='red')
plt.plot(X_test,reg.predict(X_test),color="blue")
plt.title("Sal vs Yrs Exp(testset)")
plt.ylabel("Salary")
plt.xlabel("Years Of Exp")
plt.show()